src/libraries/System.Private.CoreLib/src/System/Environment.Unix.GetFolderPathCore.cs

```diff
@@ -0,0 +1,250 @@
+                            // XDG_PICTURES_DIR = "/absolute/path"
+
+                            // Skip past whitespace at beginning of line
+                            int pos = 0;
+                            SkipWhitespace(line, ref pos);
+                            if (pos >= line.Length) continue;
+
+                            // Skip past requested key name
+                            if (string.CompareOrdinal(line, pos, key, 0, key.Length) != 0) continue;
+                            pos += key.Length;
+
+                            // Skip past whitespace and past '='
+                            SkipWhitespace(line, ref pos);
+                            if (pos >= line.Length - 4 || line[pos] != '=') continue; // 4 for ="" and at least one char between quotes
+                            pos++; // skip past '='
+
+                            // Skip past whitespace and past first quote
+                            SkipWhitespace(line, ref pos);
+                            if (pos >= line.Length - 3 || line[pos] != '"') continue; // 3 for "" and at least one char between quotes
+                            pos++; // skip past opening '"'
+
+                            // Skip past relative prefix if one exists
+                            bool relativeToHome = false;
+                            const string RelativeToHomePrefix = "$HOME/";
+                            if (string.CompareOrdinal(line, pos, RelativeToHomePrefix, 0, RelativeToHomePrefix.Length) == 0)
+                            {
+                                relativeToHome = true;
+                                pos += RelativeToHomePrefix.Length;
+                            }
+                            else if (line[pos] != '/') // if not relative to home, must be absolute path
+                            {
+                                continue;
+                            }
+
+                            // Find end of path
+                            int endPos = line.IndexOf('"', pos);
+                            if (endPos <= pos) continue;
+
+                            // Got we need.  Now extract it.
+                            string path = line.Substring(pos, endPos - pos);
+                            return relativeToHome ?
+                                Path.Combine(homeDir, path) :
+                                path;
+                        }
+                    }
+                }
+                catch (Exception exc)
+                {
+                    // assembly not found, file not found, errors reading file, etc. Just eat everything.
+                    Debug.Fail($"Failed reading {userDirsPath}: {exc}");
+                }
+            }
+
+            return Path.Combine(homeDir, fallback);
+        }
+
+        private static void SkipWhitespace(string line, ref int pos)
+        {
+            while (pos < line.Length && char.IsWhiteSpace(line[pos])) pos++;
+        }
+    }
+}
```

Comment:
None of the code here was modified, it just moved files, right?

Reply:
Yes. Could you please approve it if it looks ok, thanks